File: testdata/precision-suite-cpp/clean_placeholders.c
// Clean: placeholder credentials and a fixed banner. The strings look
// secret-shaped but are obvious placeholders in ordinary string literals, and no
// untrusted input reaches any sink. A correct scanner emits nothing; a taint
// finding here would be a false positive.
#include <stdio.h>

#define DB_USER "REPLACE_ME_USER"
#define DB_PASS "CHANGE_ME_BEFORE_DEPLOY"
#define API_KEY "your-api-key-here-000000000000"

// A generated build banner, printed with a fixed format.
static const char *kBanner =
    "app v1.0.0 (build 00000000) - configure credentials in /etc/app/secrets";

void print_banner(void) {
    printf("%s\n", kBanner);
}

File: testdata/precision-suite-rust/clean_placeholders.rs
// Clean: placeholder / example credentials and constants — the kind of strings
// broad secret rules over-fire on. None of these are live secrets, and no
// untrusted value reaches a sink. A correct scanner emits nothing; any finding
// here is a false positive.

// Example values only — documented placeholders, not real credentials.
const EXAMPLE_API_KEY: &str = "your-api-key-here";
const EXAMPLE_TOKEN: &str = "xxxxxxxxxxxxxxxxxxxxxxxx";
const DATABASE_URL_TEMPLATE: &str = "postgres://USER:PASSWORD@localhost:5432/db";

// A UUID and a hex color — high-entropy-looking but plainly not secrets.
const REQUEST_ID: &str = "550e8400-e29b-41d4-a716-446655440000";
const BRAND_COLOR: &str = "#1a2b3c";

fn describe() -> String {
    // Constant, non-tainted SQL against a constant id: safe.
    let sql = "SELECT 1";
    format!("{} key={} token={}", sql, EXAMPLE_API_KEY, EXAMPLE_TOKEN)
}
